tools/auth/notion.go

```diff
@@ -0,0 +1,88 @@
+package auth
+
+import (
+	"context"
+	"encoding/json"
+	"golang.org/x/oauth2"
+	"net/http"
+)
+
+var _ Provider = (*Notion)(nil)
+
+// NameNotion is the unique name of the Patreon provider.
+const NameNotion string = "notion"
+
+// Notion allows authentication via Notion OAuth2.
+type Notion struct {
+	*baseProvider
+}
+
+// NewNotionProvider creates new Notion provider instance with some defaults.
+func NewNotionProvider() *Notion {
+	return &Notion{&baseProvider{
+		ctx:        context.Background(),
+		authUrl:    "https://api.notion.com/v1/oauth/authorize",
+		tokenUrl:   "https://api.notion.com/v1/oauth/token",
+		userApiUrl: "https://api.notion.com/v1/users/me",
+	}}
+}
+
+// FetchAuthUser returns an AuthUser instance based on the Notion's User api.
+// API reference: https://developers.notion.com/reference/get-self
+func (p *Notion) FetchAuthUser(token *oauth2.Token) (*AuthUser, error) {
+	req, err := http.NewRequest("GET", p.userApiUrl, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	// See https://developers.notion.com/reference/versioning where header is required for all requests
+	req.Header.Set("Notion-Version", "2022-06-28")
+
+	data, err := p.sendRawUserDataRequest(req, token)
+	if err != nil {
+		return nil, err
+	}
+
+	rawUser := map[string]any{}
+	if err := json.Unmarshal(data, &rawUser); err != nil {
+		return nil, err
+	}
+
+	extracted := struct {
+		AvatarUrl string `json:"avatar_url"`
+		Bot       struct {
+			Owner struct {
+				Type string `json:"type"`
+				User struct {
+					AvatarUrl string `json:"avatar_url"`
+					Id        string `json:"id"`
+					Name      string `json:"name"`
+					Person    struct {
+						Email string `json:"email"`
+					} `json:"person"`
+				} `json:"user"`
+			} `json:"owner"`
+			WorkspaceName string `json:"workspace_name"`
+		} `json:"bot"`
+		Id        string `json:"id"`
+		Name      string `json:"name"`
+		Object    string `json:"object"`
+		RequestId string `json:"request_id"`
+		Type      string `json:"type"`
+	}{}
+
+	if err := json.Unmarshal(data, &extracted); err != nil {
+		return nil, err
+	}
+
+	user := &AuthUser{
+		Id:          extracted.Bot.Owner.User.Id,
+		Name:        extracted.Bot.Owner.User.Name,
+		Email:       extracted.Bot.Owner.User.Person.Email,
```

Comment:
Please check if `extracted.Bot.Owner.User.Person.Email` returns only **verified/confirmed** email.

If Notion allows users with unverified emails and there is no field to indicate its verified state then skip the `AuthUser.Email` assignment.

Reply:
From what I've observed, you cannot create an account in Notion unless you do OAuth or email verification.

Similarly if you decide later change the email, you have to verify the new one too

See:
![image](https://github.com/pocketbase/pocketbase/assets/62984456/6111122d-0808-44ac-9c0a-6b8bee199ea0)
![image](https://github.com/pocketbase/pocketbase/assets/62984456/ab89503b-614a-4923-a815-16d1397dde78)


Does leaving it as it is suffice?

Reply:
In this case I think it is fine. Thank you for checking.